Add feature: Go to definition

QuangNguyenMinh123 opened the pull request:
This MR adds a “Go to Definition” feature, mimicking VS Code
In which, I will use 2 short cuts: "F12" and "Ctrl + left click"
<img width="551" height="255" alt="image" src="https://github.com/user-attachments/assets/949e3ab6-171b-4f7f-bd47-0896c864a81d" />

epasveer commented:
> @epasveer What do you think about this MR?

Apologies. I've been busy with the Holidays and New Years.  I'll review this MR shortly.

I hope you've had a good Holidays.

epasveer commented:
Okay.  I'll review it again when you're done.

QuangNguyenMinh123 commented:
Hi @epasveer ,  sorry for delay
Since types, variables, and functions rarely have exactly the same name, could we just temporarily ignore this case, mark it as a known issue, and merge this MR anyway?
I use my Seer version with OpenOCD to debug embedded systems at my workplace, and I’ve actually never encountered a case where identifiers had identical names.

QuangNguyenMinh123 committed:
SeerSourceBrowserWidget: add findFileWithRegrex

QuangNguyenMinh123 committed:
Add Go to Definition feature

QuangNguyenMinh123 commented:
Ready for review, please check

epasveer commented:
To confirm, there is no Menu Item change to invoke the 'Go to definition' call.  Just a F12 hotkey?

QuangNguyenMinh123 commented:
I fixed it, please help me to review
One thing to note is that I added a function called findFileWithRegex to SeerSourceBrowserWidget.
This function checks whether a file exists in the list of source code files.
This function is not crucial when debugging MCUs, since the source code size is usually small. However, it plays an important role when debugging Linux-based embedded systems. Because Linux source code is very large, this function can drastically reduce parsing time.

epasveer committed:
Made tweaks to this PR.

QuangNguyenMinh123 committed:
SeerEditorManagerWidget: Eliminate noise values

epasveer committed:
Handle when function symbols have or don't have arguments.

epasveer committed:
Removed SeerEditorManagerWidget knowing SeerSourceBrowserWidget. Not needed.

epasveer committed:
Clean up SeerSourceBrowserWidget.

epasveer committed:
More cleanup.

QuangNguyenMinh123 commented:
Hi @epasveer,

> I see you removed _sourceBrowserWidget and findFileWithRegex.
> These might not be useful right now, but they will be when it comes to debugging the Linux kernel, where the source code is massive

As mentioned earlier, I planned to implement findFileWithRegex to speed up "Go to Definition".
I tried it, but it turns out there's no noticeable improvement, so I don't think it's worth implementing.
